src/tests/conformance/query/peer_shard_info.rs

```diff
@@ -24,3 +35,86 @@ async fn c006_TM_GET_PEER_SHARD_INFO_V2_node_should_not_query_for_shard_info_if_
         |m: &BinaryMessage| matches!(&m.payload, Payload::TmGetPeerShardInfoV2(..));
     perform_response_test(TestConfig::default().with_handshake(false), &response_check).await;
 }
+
+#[tokio::test]
+#[allow(non_snake_case)]
+async fn c011_TM_GET_PEER_SHARD_INFO_V2_node_should_relay_shard_info() {
+    // ZG-CONFORMANCE-011
+    for key_type in KEY_TYPES {
+        check_relay_for_key_type(*key_type, 2).await;
+    }
+}
+
+#[tokio::test]
+#[should_panic]
+#[allow(non_snake_case)]
+async fn c012_TM_GET_PEER_SHARD_INFO_V2_node_should_not_relay_shard_info_with_invalid_key_type() {
+    // ZG-CONFORMANCE-012
+    check_relay_for_key_type(0x42, 2).await;
+}
+
+#[tokio::test]
+#[should_panic]
+#[allow(non_snake_case)]
+async fn c013_TM_GET_PEER_SHARD_INFO_V2_node_should_not_relay_shard_info_when_relays_equals_zero() {
+    // ZG-CONFORMANCE-013
+    check_relay_for_key_type(KEY_TYPES[0], 0).await;
+}
+
+#[tokio::test]
+#[should_panic]
+#[allow(non_snake_case)]
+async fn c014_TM_GET_PEER_SHARD_INFO_V2_node_should_not_relay_shard_info_when_relays_above_limit() {
+    // ZG-CONFORMANCE-014
+    check_relay_for_key_type(KEY_TYPES[0], 4).await;
```

Comment:
Can we mention here again with a comment that the limit relay_limit is 3?
Or use a const maybe:

```
    const RELAY_LIMIT: usize = 3;
    check_relay_for_key_type(KEY_TYPES[0], RELAY_LIMIT + 1).await;
```

Reply:
Done